### PR TITLE
Feature user auth0

### PR DESCRIPTION
**Pull request by aguspina**

### CODE REVIEW

File: lib/mumukit/auth/user.rb [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    client.update_user_metadata social_id, metadata.merge(data)
+  end
+
+  def metadata
+    metadata = {}
+    apps.each do |app|
+      metadata[app] = @user[app] if @user[app].present?
+    end
+    metadata
+  end
```
Comment (flbulgarelli) [inline lib/mumukit/auth/user.rb:28]: This is a map + select

**Comment by aguspina** [comment]
@flbulgarelli  please review

File: lib/mumukit/auth/user.rb [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+  def permissions_for(app)
+    @metadata[app]['permissions']
+  end
+
+  def apps
+    ['bibliotheca', 'classroom', 'admin', 'atheneum']
+  end
+
+  def can_visit?(app)
```
Comment (flbulgarelli) [inline lib/mumukit/auth/user.rb]: I would put this in Atheneum, not here.

File: lib/mumukit/auth/user.rb [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    @metadata = get_metadata || {}
+  end
+
+  def update_permissions(key, permission)
+    add_permission!(key, permission)
+    client.update_user_metadata social_id, @metadata
+  end
+
+  def get_metadata
+    apps.select { |app| @user[app].present? }.map { |app| { "#{app}" => @user[app] } }.reduce({}, :merge)
```
Comment (flbulgarelli) [inline lib/mumukit/auth/user.rb]: use `app.to_s` instead of interpolation

Comment (flbulgarelli) [inline lib/mumukit/auth/user.rb]: I'd rather call this method metadata and cache it internally in an attribute 

```
def metadata
   @metadata ||= (apps.select { |app| @user[app] ... .|| {})
```

**Comment by flbulgarelli** [comment]
Please review my comments and merge

File: lib/mumukit/auth/user.rb [hunk 4]
```diff
@@ -18,7 +18,7 @@ def get_metadata
     apps.select { |app| @user[app].present? }.map { |app| { "#{app}" => @user[app] } }.reduce({}, :merge)
   end
```
Comment (flbulgarelli) [inline lib/mumukit/auth/user.rb:20]: This commit is named `foo`. Wtf?

**Pull request merged**